### ISSUE TITLE
Autocheck for updates using github api

### ISSUE THREAD

**Issue by AsherGlick** [AsherGlick/Burrito#37]
We can use the public github API to check to see if there has been a newer release then the one currently being used. In order to do this some release metadata will need to be saved into the distribution itself, and a new piece of code that runs when burrito launches will need to query the github release API endpoint https://api.github.com/repos/AsherGlick/Burrito/releases to see if there are newer releases.

The API lists newest releases first and at the time of writing seems to return the newest 30 releases. I think something as simple as saving the release name on automatic build, and checking if that release name against the newest version will be sufficient. This check will not be performed on builds that do not have a release variable set so that local development does not run into issues where their local build is not equivalent to the newest version. Versioning will be as follows. An `alpha-` prefix will denote an earlier version then a `beta-` prefix or no prefix. A `beta-` prefix will denote an earlier version then no prefix. Finally no prefix will have a `major` and `minor` component.

EG:
`alpha-1.2`
`alpha-1.3`
`alpha-1.4`
`beta-1.0`
`beta-1.1`
`beta-2.3`
`1.0`
`1.1`
`1.2`

**Comment by coderedart** [comment]
If burrito has some kind of config file, then there can be an option to autoupdate - yes/no inside that file.

**Comment by AsherGlick** [comment]
Good idea. What I had in mind for this issue is just checking for updates not actually updating, but if a ci system is able to take over pushing updates then an auto update system could be built in too. I agree it should be use configurable.

**Comment by royalmustard** [comment]
maybe we should do it in a way where it tells you there is an update available, but you still have to confirm to update (or not update)

**Comment by logan001** [comment]
also make sure to add a downgrade option too in case there are bugs with the newer version
maybe a nice way is to make a scroll box with every version and select the one you want (see how lutris lets you select dxvk version including add your own version manually in that text box)

**Comment by coderedart** [comment]
from what i understand, there's only two ways this can work. either download the new executable and replace it with the current one and ask the user to restart. self-restart seems to be a complicated thing (I have been looking into it ) with execv and such syscalls, so asking the user to restart might be easier to start with. 

 or have a seperate launcher app which will check for updates and then start the main overlay executable.

**Comment by AsherGlick** [comment]
I see "checking for updates automatically" and "automatically updating" as two separate features, with the latter being significantly more complicated. Checking for updates alone would not require features such as a version management pre-launcher or auto restart as users would be taking care of the upgrade / downgrade process manually.
